### PR TITLE
DataNode: Adding the "restart graylog" step, removed unused enum steps

### PR DESCRIPTION
**Pull request by janheise**
## Description

We have to ask the user to restart graylog to finish the migration.

/nocl

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog/plugins/views/storage/migration/state/machine/MigrationStateMachineBuilder.java [hunk 1]
```diff
@@ -73,6 +73,7 @@ private static StateMachineConfig<MigrationState, MigrationStep> configureStates
         config.configure(MigrationState.PROVISION_DATANODE_CERTIFICATES_PAGE)
                 .permit(MigrationStep.PROVISION_DATANODE_CERTIFICATES, MigrationState.PROVISION_DATANODE_CERTIFICATES_RUNNING, migrationActions::provisionAndStartDataNodes);
 
+        // This page should contain the "Please restart Graylog to continue with data migration"
```
Comment (moesterheld) [inline graylog2-server/src/main/java/org/graylog/plugins/views/storage/migration/state/machine/MigrationStateMachineBuilder.java:76]: Remote Reindexing isn't going to work this way, I am afraid. 
After provisioning is finished, the connection string needs to be removed and Graylog needs to be restarted to connect to the data nodes.

Comment (moesterheld) [reply graylog2-server/src/main/java/org/graylog/plugins/views/storage/migration/state/machine/MigrationStateMachineBuilder.java:76]: this should work now after the change i pushed. we just have to make the user aware that he needs to remove the connection string when we tell him to restart

**Comment by moesterheld** [comment]
I did some cleanup
@janheise or @todvora  please look over it once more

**Comment by janheise** [comment]
@moesterheld thx for the cleanup. imho everything should be ok now - waiting for @todvora 's review

**Review by todvora — APPROVED** [review]
LGTM!

**Pull request merged**